legacy/builder/builder_utils/utils.go

```diff
@@ -153,17 +67,35 @@ func findAllFilesInFolder(sourcePath string, recurse bool) ([]string, error) {
 	return sources, nil
 }
 
-func compileFilesWithRecipe(ctx *types.Context, sourcePath *paths.Path, sources paths.PathList, buildPath *paths.Path, buildProperties *properties.Map, includes []string, recipe string) (paths.PathList, error) {
+func CompileFiles(ctx *types.Context, sourcePath *paths.Path, recurse bool, buildPath *paths.Path, buildProperties *properties.Map, includes []string) (paths.PathList, error) {
+	var sources paths.PathList
+	var err error
+	if recurse {
+		sources, err = sourcePath.ReadDirRecursive()
+	} else {
+		sources, err = sourcePath.ReadDir()
+	}
+	if err != nil {
+		return nil, err
+	}
+
+	validExtensions := []string{".S", ".c", ".cpp"}
+
+	sources.FilterSuffix(validExtensions...)
+	ctx.Progress.AddSubSteps(len(sources))
+	defer ctx.Progress.RemoveSubSteps()
+
 	objectFiles := paths.NewPathList()
+	var objectFilesMux sync.Mutex
```

Comment:
Why move this declaration? :thinking:

Reply:
puts the mutex near the guarded object, just a small readability improvement

Reply:
Ah, makes sense. Didn't notice.